Feat: Base interface & documentation (#1)
dashevo/DashJS: README.md

    network,
    mnemonic: "arena light cheap control apple buffalo indicate rare motor valid accident isolate",
    schema
};
const sdk = new DashJS.SDK(opts);
const acc = sdk.wallet.getAccount();
async function sendPayment(){
    const tx = await acc.createTransaction({recipient:{address:'yLptqWxjgTxtwKJuLHoGY222NnoeqYuN8h', amount:0.12}})
    console.log(tx)
}

async function readDocument() {
    const profile = await sdk.platform.fetchDocuments('profile',{},opts)
    console.log(profile);
}
```

Notes : 

- Omitting mnemonic will set the Wallet functionalities in offlineMode (for resources savings purposes) and set a default mnemonic.  
 You can use `sdk.wallet.exportWallet()` to get the randomly generated mnemonic.
- Omitting a schema will unset the Platform functionalities.

### Platform

Access the [Platform documentation on dashevo/dashcore-lib](/docs/Platform.md)

### Wallet

Access the [Wallet documentation on dashevo/dashcore-lib](/docs/Wallet.md)

### Primitives

In order to facilitate manipulations between the dependencies, DashJS provides the standardized Primitives as implemented in [Dashcore-lib](https://github.com/dashevo/dashcore-lib).

#### Transaction 

> The Transaction primitives allow easy creation and manipulation of transaction. It also allow signing when provided a privatekey.  

Reviewer: ```suggestion
The Transaction primitive allows easy creation and manipulation of transactions. It also allows signing when provided with a privatekey.  
```

> Fee control and input/output access provided (which allow to pass along specific script).  

Reviewer: ```suggestion
Supports fee control and input/output access (which allows to pass a specific script).  
```

```js
import { Transaction } from '@dashevo/dashjs';
const tx = new Transaction(txProps)
```

Access the [Transaction documentation on dashevo/dashcore-lib](https://github.com/dashevo/dashcore-lib/blob/master/docs/transaction.md)

#### Address

Standardized representation of a Dash Address. Address can be instantiated from a String, PrivateKey, PublicKey, HDPrivateKey or HdPublicKey.  

> It also support pay-to-script-hash multi-sinatures from an array of PublicKeys.  

Reviewer: ```suggestion
pay-to-script-hash multi-signatures from an array of PublicKeys are also supported.  
```

```js
import { Address } from '@dashevo/dashjs';
```

Access the [Address documentation on dashevo/dashcore-lib](https://github.com/dashevo/dashcore-lib/blob/master/docs/address.md)

#### Block

> Given an hexadecimal string representation of the block as input, the Block class allow you to have a deserialized representation of a Block, it's header and can validate it.  

Reviewer: ```suggestion
Given a hexadecimal string representation of the block as input, the Block class allows you to have a deserialized representation of a Block or it's header. It also allows to validate the transactions in the block against the header merkle root.  
```

> Transaction of the block can also be explorated by iterating on transactions member (`block.transactions`).

Reviewer: ```suggestion
Transactions of the block can also be explored by iterating over elements in array (`block.transactions`).
```

# DashJS

[![Package Version](https://img.shields.io/github/package-json/v/dashevo/dashjs.svg?&style=flat-square)](https://www.npmjs.org/package/@dashevo/dashjs)
[![Build Status](https://img.shields.io/travis/com/dashevo/dashjs.svg?branch=master&style=flat-square)](https://travis-ci.com/dashevo/dashjs)

> Dash library for JavaScript/TypeScript ecosystem (Wallet, DAPI, Primitives, BLS, ...)

## Table of Contents

- [State](#state)
- [Principles](#principles)
- [Install](#install)
- [Usage](#usage)
- [License](#license)

## State

Under active development.

## Principles

Dash is a powerful new peer-to-peer platform for the next generation of financial technology. The decentralized nature of the Dash network allows for highly resilient Dash infrastructure, and the developer community needs reliable, open-source tools to implement Dash apps and services.

## Install

```sh

> npm install @dashevo/dashjs

Reviewer: I thought we especially secured https://www.npmjs.com/package/dash and were going to use it for this project?

Author: Using @dashevo has a benefit on the configuration of it (it's under a npm org account). 
What do you suggest ? Switch on /dash ? Mirroring seems not a possible options :x

Reviewer: @dalibran @shumkov @antouhou as I remember the plan was all along to be able to just use
`npm install dash` for the SDK. Have there been any changes to this?

Author: Well, the whole repo is called DashJS, so even without the @dashevo org, I was expecting at least this. 
And then there is the whole maintain process having it elsewhere. 
The thing we really did not want was a 4yr old fork of bitpay or whatever it was under the name dash as one doing npm i dash might end up downloading something that is not working. 
So may initial guess would to put version there with a deprecate message leading to a @dashevo owned npm ?

Reviewer 2: The plan is to still use `npm install dash` and refer to all of the client libraries as the Dash SDK. I'd prefer to not distinguish between the other implementations by different names (i.e. DashJS or DashJ or go-dash), since they should all of the same interface and therefore same developer experience. 

We've discussed naming them using the same convention in Github so that they appear alphabetically for easier searching -- something like `dash-js`, `dash-go`, `dash-py`

@nmarley can provide more context as to the best repo naming convention

Reviewer 3: We discussed and started using language prefixes because there can be multiple implementations of a library or standard, e.g. `js-dpp`, `ios-dpp`, and `android-dpp` for example. This prefix convention was copied from the IPFS org (cf `go-ipfs` and `js-ipfs`, etc.) and as a bonus makes it easier to filter language-specific repos in a search.
